### PR TITLE
feat(core): warn leading slash for output.distPath.html and default to `./`

### PR DESCRIPTION
**Pull request by wjw99830**
## Summary
Currently the default value of `output.distPath.html` is `/`. It's relative to `output.distPath.root`.
But other props(like `js`, `css`...) cannot be `/`, because they're relative to the root of file system.

## Related Links
https://rsbuild.dev/config/output/dist-path

## Checklist

- [x] Tests updated (or not required).
- [x] Documentation updated (or not required).

### CODE REVIEW

**Review by chenjiahan — COMMENTED** [review]
I think `output.distPath.html` defaults to `/` is a mistake, it should default to `./`.

By design, `output.distPath.*` can only be relative (except for `output.distPath.root`). If we allow all properties to start with `/`,  I would worry that it might confuse some users.

**Comment by wjw99830** [comment]
> I think `output.distPath.html` defaults to `/` is a mistake, it should default to `./`.
Yeah, i think so too.
If we modify the default value to `./`, it will be a breaking change for user.

**Comment by chenjiahan** [comment]
Yes, we don't want to introduce breaking changes, we can maintain compatibility with the previous behavior and print a warning if the user passes an absolute path to `distPath.html`

File: packages/core/src/constants.ts [hunk 1]
```diff
@@ -3,7 +3,7 @@ import { join } from 'node:path';
 // Paths
 // loaders will be emitted to the same folder of the main bundle
 export const ROOT_DIST_DIR = 'dist';
-export const HTML_DIST_DIR = '/';
+export const HTML_DIST_DIR = '';
```
Comment (chenjiahan) [inline packages/core/src/constants.ts:6]: can we use `./` as the default value? It looks more explicit

**Review by chenjiahan — APPROVED** [review]
Thanks!

**Pull request merged**